packages/aws-cdk/lib/api/deploy-stack.ts

```diff
@@ -151,9 +151,12 @@ export async function deployStack(options: DeployStackOptions): Promise<DeploySt
     // tslint:disable-next-line:max-line-length
     const monitor = options.quiet ? undefined : new StackActivityMonitor(cfn, deployName, options.stack, (changeSetDescription.Changes || []).length).start();
     debug('Execution of changeset %s on stack %s has started; waiting for the update to complete...', changeSetName, deployName);
-    await waitForStack(cfn, deployName);
-    if (monitor) {
-      await monitor.stop();
+    try {
+      await waitForStack(cfn, deployName);
+    } finally {
+      if (monitor) {
+        await monitor.stop();
```

Comment:
Not that you need to change it, but I think we can forego the `if` and write`

```ts
await monitor?.stop();
```

These days...